appserver/payara-appserver-modules/payara-micro-service/src/main/java/fish/payara/appserver/micro/services/PayaraInstanceImpl.java

```diff
@@ -242,6 +247,12 @@ public void event(Event event) {
                 me.addApplication(new ApplicationDescriptorImpl(appRegistry.get(appName)));
             }
             cluster.getClusteredStore().set(INSTANCE_STORE_NAME, myCurrentID, me);
+            executor.scheduleAtFixedRate(() -> {
```

Comment:
Shouldn't the `ScheduledFuture` returned by this method be stored with this service as a member and be cancelled when this service is shutdown? Somewhere in a `@PreDestroy` annotated method of this service?